add talk api client to get subject discussions

camallen opened the pull request:
Linked to https://github.com/zooniverse/Panoptes/pull/1833 

Add the panoptes talk client to find subjects with discussions. I'd like to refactor the panoptes client base functionality with composition but wanted to get a first pass by @marten for comments.

marten commented on lib/panoptes/client/discussions.rb:
```diff
@@ -0,0 +1,19 @@
+        query[:focus_id] = focus_id
+        query[:focus_type] = focus_type
+
+        response = get("/discussions", query)
```
It's an authenticated call this way right? Will that return discussions for private projects?

marten replied:
Meant to say **un**authenticated

camallen replied:
figured - not sure at all to be honest.

camallen commented:
@marten updated to add authenticated API req's for a talk client. Extracted the common client config / setup and added an `auth_url` param for talk login to different api url.

Lots of hound issues, happy to clean up.
